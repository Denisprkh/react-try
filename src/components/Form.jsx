import React from "react";
import configData from "../config/config.json";

class Form extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      price: "",
      description: "",
      duration: "",
      tags: [],
    };
    this.handleNameChange = this.handleInputChange.bind(this);
    this.handleSumbit = this.handleSumbit.bind(this);
  }

  handleSumbit(event) {
    event.preventDefault();
    this.createCertificate();
  }

  async createCertificate() {
    const postData = {
      method: "POST",
      headers: {
        "Content-Type": "application/json",
        Accept: "application/json",
      },
      body: JSON.stringify({
        price: this.state.price,
        description: this.state.description,
        name: this.state.name,
        duration: this.state.duration,
        tags: this.state.tags.split(" "),
      }),
    };
    const response = await fetch(configData.API_URL, postData);
  }

  handleInputChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  };

  render() {
    const { name, price, description, duration, tags } = this.state;
    return (
      <form onSubmit={this.handleSumbit}>
        <input
          name={"name"}
          placeholder="name"
          value={name}
          onChange={this.handleInputChange}
        />
        <input
          name={"price"}
          placeholder="price"
          value={price}
          onChange={this.handleInputChange}
        />
        <input
          name={"description"}
          placeholder="description"
          value={description}
          onChange={this.handleInputChange}
        />
        <input
          name={"duration"}
          placeholder="duration"
          value={duration}
          onChange={this.handleInputChange}
        />
        <input
          name={"tags"}
          placeholder="tags"
          value={tags}
          onChange={this.handleInputChange}
        />
        <input type="submit" />
      </form>
    );
  }
}

export default Form;
