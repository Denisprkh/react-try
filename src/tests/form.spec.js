import { shallow } from 'enzyme';
import React from 'react';
import Form from '../components/Form';
import testConfig from '../config/testConfig.js';

describe("should render Form component", () => {
    it("should contain .creation-form wrapper", () => {
        const component = shallow(<Form/>);
        const wrapper = component.find(".creation-form");
        expect(wrapper.length).toBe(1);
    });
    
    it("should match snapshot", () => {
        const component = shallow(<Form/>);
        expect(component).toMatchSnapshot();
    });
});
