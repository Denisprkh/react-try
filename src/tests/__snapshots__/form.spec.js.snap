// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sds  1`] = `ShallowWrapper {}`;

exports[`should render Form component should match snapshot 1`] = `ShallowWrapper {}`;
